floodviz/static/js/map.js

```diff
@@ -1,187 +1,321 @@
+				.attr("id", "maptip");
+
+			/**
+			 * Add circles to the map.
+			 * @param data The geojson to be added to the svg
+			 * @param classname The class to be given to each element for use in CSS
+			 * @param radius The radius of each circle. This cannot be set from CSS
+			 * @param property_for_id The name of a field in the 'properties' of each feature, to be used for ID
+			 *                            If null, or not provided, no id will be given.
+			 */
+			var add_circles = function (data, classname, radius, property_for_id) {
+				var group = svg.append("g");
+				group.selectAll("circle")
+					.data(data.features)
+					.enter()
+					.append("circle")
+					.attr("r", radius)
+					.attr("transform", function (d) {
+						return "translate(" + projection(d.geometry.coordinates) + ")";
+					})
+					.attr("id", function (d) {
+						if (property_for_id && d.properties[property_for_id]) {
+							return 'map' + d.properties[property_for_id];
 						}
-					}
+						else {
+							return '';
+						}
+					})
+					.attr("class", classname)
+					.each(function (d) {
+						if (property_for_id && d.properties[property_for_id]) {
+							if (FV.hydrograph_display_ids.indexOf(d.properties.id) !== -1) {
+								self.site_add_accent(d.properties.id);
+							}
+						}
+					});
+				return (group);
+			};
+			/**
+			 * Add paths to the map
+			 * @param data The geojson to be added to the svg
+			 * @param classname The class to be given to each element for use in CSS
+			 */
+			var add_paths = function (data, classname) {
+				var group = svg.append("g");
+				group.selectAll("path")
+					.data(data.features)
+					.enter()
+					.append("path")
+					.attr("d", path)
+					.attr("class", classname);
+				return group;
+			};
+
+			/**
+			 * Initialize the Map
+			 */
+			self.init = function () {
+
+				if (svg !== null) {
+					d3.select(options.div_id).select('svg').remove();
+				}
+				svg = d3.select(options.div_id)
+					.append("svg")
+					.attr("width", options.width)
+					.attr("height", options.height);
+
+				// Define the drag behavior to be used for the selection box
+				var drag = d3.drag()
+					.on('start', function () {
+						var p = d3.mouse(this);
+						select_box_start(p);
+					})
+					.on('drag', function () {
+						var p = d3.mouse(this);
+						select_box_drag(p);
+					})
+					.on('end', function(){
+						select_box_end();
+					});
+
+
+				// set bounding box to values provided
+				var b = path.bounds(options.bounds);
+				var s = options.scale / Math.max((b[1][0] - b[0][0]) / options.width, (b[1][1] - b[0][1]) / options.height);
+				var t = [(options.width - s * (b[1][0] + b[0][0])) / 2, (options.height - s * (b[1][1] + b[0][1])) / 2];
+				// Update the projection
+				projection.scale(s).translate(t);
+				// Add layers
+				var bg =add_paths(options.bg_data, "background");
+				bg.call(drag);
+				var rivers = add_paths(options.rivers_data, "river");
+				rivers.call(drag);
```

Comment:
It lets the user click and drag on a river, which will be nice if the rivers are thicker than they are in the Iowa example.